cordova-lib/src/plugman/install.js

```diff
@@ -79,21 +79,36 @@ module.exports = function installPlugin(platform, project_dir, id, plugins_dir,
     }
 
     var current_stack = new action_stack();
-
     // Split @Version from the plugin id if it exists.
     var splitVersion = id.split('@');
     //Check if a mapping exists for the plugin id
     //if it does, convert id to new name id
-    var newId = pluginMapper[splitVersion[0]];
+    var newId = pluginMapper.oldToNew[splitVersion[0]];
     if(newId) {
-        events.emit('warn', 'Notice: ' + id + ' has been automatically converted to ' + newId + ' and fetched from npm. This is due to our old plugins registry shutting down.');
         if(splitVersion[1]) {
             id = newId +'@'+splitVersion[1];
         } else {
             id = newId;
         }
-     }
-    return possiblyFetch(id, plugins_dir, options)
+    }
+    var P;
+    var plugin_dir = path.join(plugins_dir, splitVersion[0]);
+    // if the plugin has already been fetched, use it.
+    if (fs.existsSync(plugin_dir)) {
+        P = Q(plugin_dir);
+    } else {
+        var alias = pluginMapper.newToOld[splitVersion[0]] || newId;
+        // if the plugin alias has already been fetched, use it.
+        if (alias && fs.existsSync(path.join(plugins_dir, alias))) {
+            P = Q(path.join(plugins_dir, alias));
```

Comment:
Maybe we should show a message when we're going to use already fetched plugin with old id despite that user tries to install the one with new id?